src/Microsoft.AspNetCore.Razor.Language/RazorLanguageVersion.cs

```diff
@@ -2,32 +2,107 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Diagnostics;
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
-    public sealed class RazorLanguageVersion : IEquatable<RazorLanguageVersion>
+    [DebuggerDisplay("{" + nameof(DebuggerToString) + "(),nq}")]
+    public sealed class RazorLanguageVersion : IEquatable<RazorLanguageVersion>, IComparable<RazorLanguageVersion>
     {
-        public static readonly RazorLanguageVersion Version_1_0 = new RazorLanguageVersion(1, 0);
+        public static readonly RazorLanguageVersion Version_1_0 = new RazorLanguageVersion("1.0", 1, 0);
 
-        public static readonly RazorLanguageVersion Version_1_1 = new RazorLanguageVersion(1, 1);
+        public static readonly RazorLanguageVersion Version_1_1 = new RazorLanguageVersion("1.1", 1, 1);
 
-        public static readonly RazorLanguageVersion Version_2_0 = new RazorLanguageVersion(2, 0);
+        public static readonly RazorLanguageVersion Version_2_0 = new RazorLanguageVersion("2.0", 2, 0);
 
-        public static readonly RazorLanguageVersion Version_2_1 = new RazorLanguageVersion(2, 1);
+        public static readonly RazorLanguageVersion Version_2_1 = new RazorLanguageVersion("2.1", 2, 1);
 
-        public static readonly RazorLanguageVersion Latest = Version_2_1;
+        public static readonly RazorLanguageVersion Latest = new RazorLanguageVersion("Latest", 2, 1);
+
+        public static bool TryParse(string languageVersion, out RazorLanguageVersion version)
+        {
+            if (languageVersion == null)
+            {
+                throw new ArgumentNullException(nameof(languageVersion));
+            }
+
+            if (languageVersion == "Latest")
+            {
+                version = Latest;
+                return true;
+            }
+            else if (languageVersion == "2.1")
+            {
+                version = Version_2_1;
+                return true; 
+            }
+            else if (languageVersion == "2.0")
+            {
+                version = Version_2_0;
+                return true;
+            }
+            else if (languageVersion == "1.1")
+            {
+                version = Version_1_1;
+                return true;
+            }
+            else if (languageVersion == "1.0")
+            {
+                version = Version_1_0;
+                return true;
+            }
+
+            version = null;
+            return false;
+        }
+
+        public static RazorLanguageVersion Parse(string languageVersion)
```

Comment:
@NTaylorMullen so aspects of your original design turned out to be problematic once parsing is introduced. Your ToString() doesn't roundtrip and if latest == 2.1 then that's a hard problem to solve.

I went a different way, open to feedback.

In this model, you use IComparable rather than equals to tell if the version supports a feature. So equality still is just reference equality.

Reply:
Hmm should we just do away with "Latest" then? We could always force people to specify the version. The only piece I don't like is that `RazorLanguageVersion.Latest != RazorLanguagVersion.Version_2_1`. I guess we could always override the equality operator.

Reply:
I want to make sure that Latest can be parsed. If you're OK with not roundtripping latest maybe your original design is OK. I might just revert all of this, let me think about it a bit more.

Reply:
This seems good. We're consistent with how Roslyn does this while being better in that you don't have to call `GetEffectiveVersion`